### PR TITLE
A lot more slide updates

### PR DESCRIPTION
**Pull request by jypma**

### CODE REVIEW

File: LB3210.org [hunk 1]
```diff
@@ -181,35 +242,143 @@ Full-text search is often important when dealing with data.
...
 TODO write about Jaeger and Zipkin
-** Configuration
-- configuration file changes
-  * kubernetes
-  * hot reloading? Not in spring boot (watch file and shutdown instead)
-- environment variables for service injection: don't do it (ordering issues)
-  * use dns instead (e.g. dns-java, akka discovery, [...])
 ** Deployment
+*** Virtualization and containerization
+- First, therer was plain hardware
```
Comment (jbendsen) [inline LB3210.org]: therer > there

File: LB3210.org [hunk 2]
```diff
@@ -181,35 +242,143 @@ Full-text search is often important when dealing with data.
...
+  * Memory and disk overhead
+  * Linux optimizations (kernel shared memory)
+- Linux can do many of this natively
+  * /Namespaces/: Hide processes from each other
+  * /Cgroups/: Limit resource usage
+- Containers to make it fast and memory-efficient
+  * Instant startup
 *** Docker
 - Limited to linux in this course
 - Lightweight layer over native cgroups isolation
```
Comment (jbendsen) [inline LB3210.org:279]: maybe you should point out that docker images are typical in 100's of megabytes where other virtualization images (e.g. virtualbox) takes up 10's of gigabytes (feel free to adjust the numbers; the point is that docker images typically is not an issue in modern environments)

File: LB3210.org [hunk 3]
```diff
@@ -181,35 +242,143 @@ Full-text search is often important when dealing with data.
...
+  * Between developers and the customer
+  * Between developers and the user
+- Is everyone speaking the same language?
+
+- /Ubiqutous language/: All team members use important terms in the same way
+  * Within a bounded context
+*** Event storming workshop
+- We need to quickly learn a new domain
+- /Business process modeling/ and /requirements gathering/
+- Bring together /domain expers/ and /developers/
```
Comment (jbendsen) [inline LB3210.org]: expers > experts

File: LB3210.org [hunk 4]
```diff
@@ -220,69 +389,207 @@ TODO write about Jaeger and Zipkin
...
+  * Client has a certificate, proving it's who it claims
+    + Server has established trust on a root certificate, having signed the client certificate
+
+- In practice
+  * Create (or purchase) a root certificate for your business, lock it tight
+  * Create intermediate CAs for particular roles, e.g. for singing micro-services
+  * Use /Certificate Signing Requests/ to reflect real business flow
+  * For your service clients
+    + Have server sign client certificates directly
+    + Or, deligate to an intermediate CA, and implement whitelisting
```
Comment (jbendsen) [inline LB3210.org]: deligate > delegate

File: LB3210.org [hunk 5]
```diff
@@ -220,69 +389,207 @@ TODO write about Jaeger and Zipkin
...
-- Spring annotation style vs. functional style (e.g. [[https://www.exoscale.com/syslog/migrate-from-hystrix-to-resilience4j/][vavr and resiliance4j]]. JOOQ defaults to lambda transactions as well.)
-   * Disadvantages of annotations: Discoverability, Composability, Testability
-- Show MVC annotations vs. akka-http lambdas
-- Resiliance4j also has [[https://resilience4j.readme.io/docs/getting-started-3][spring wrappers]]
-** Useful modern Java features
+** TODO Useful modern Java features
 - Lambdas (Java 8)
 - Records (Java 14)
   * JOOQ was [[https://github.com/jOOQ/jOOQ/issues/10287][just]] updated with record support for POJOs (for 3.15.0)
 - Type-inferrerred variables (Java 11)
```
Comment (jbendsen) [inline LB3210.org]: inferrerred > inferred

File: LB3210.org (hunk 2 above)
Comment (jypma) [reply LB3210.org:279]: Yes good one! Also good to go into ubuntu vs. alpine base images.

File: LB3210.org [hunk 6]
```diff
@@ -220,69 +389,207 @@ TODO write about Jaeger and Zipkin
...
+- Rate of bugs introduced into systems are a function of
+  * Developer experience
+  * Development environment (physical and technological)
+  * Methodology
+
+- Fixing bugs is more expensive, the later they are found
+  * While writing code: just think of different solution
+  * While code is in review: communication, context switch, and the above
+  * While code is in user testing: (much) more communication, context switch, and all the above
+  * After code is released: (even) more communication, impact anallysis in data, and all the above
```
Comment (jbendsen) [inline LB3210.org]: anallysis > analysis

**Review by jbendsen — COMMENTED** [review]
generally fine! I've pointed out some typos.

**Pull request merged**